### PR TITLE
ci/deploy: Add information box about additional pushed tags

### PR DESCRIPTION
**Pull request by antifuchs**
This PR adds another informational message to buildkite builds, writing the names of any multi-arch docker tags, and adds a build step that adds `devel-<SHA>` tags to any commit that gets built in buildkite (so that it becomes possible to spin up cloud platform environments from a pull request).

### Motivation

   * This PR refactors existing code.

Allows us to more easily spin up "test" environments in our cloud platform testing.

### Testing

The results of this should be very visible on the buildkite page for this PR's head commit. I am making it a draft for now, hope that something useful gets pushed.

### Release notes

No user-visible changes

### CODE REVIEW

**Comment by benesch** [comment]
Er, I'm confused, @antifuchs! That job only runs on merge to main. Don't think you're going to see anything on the PR build?

**Comment by antifuchs** [comment]
Yep, I just noticed that myself :D

The follow-on change should add an annotation to the buildkite build page (with a `devel-<SHA>` tag).

File: ci/test/build.py [hunk 1]
```diff
@@ -29,6 +29,7 @@ def main() -> None:
     print("--- Acquiring mzbuild images")
     deps = repo.resolve_dependencies(image for image in repo if image.publish)
     deps.ensure()
+    mzbuild.publish_multiarch_images(f'devel-{git.rev_parse("HEAD")}', deps)
```
Comment (benesch) [inline ci/test/build.py:32]: Ah, shoot, this is not going to be so simple. `publish_multiarch_images` needs _two_ dependency sets: one for aarch64 and one for x86_64. It looks like we'll need a separate job in the build pipeline that waits on both the x86_64 and aarch64 build jobs, then constructs the two dep sets, and _then_ publishes the multiarch images.

Comment (antifuchs) [reply ci/test/build.py:33]: Hm, can't this be done incrementally, as two pushes? I felt like that might be possible...

Comment (antifuchs) [reply ci/test/build.py:33]: Anyway, additional build step it is (:

Comment (benesch) [reply ci/test/build.py:33]: > Hm, can't this be done incrementally, as two pushes? I felt like that might be possible...

Not easily, no! The manifest you push up to Docker Hub needs to have both images inside of it.

**Pull request merged**